prometheus/handler.go

```diff
@@ -0,0 +1,136 @@
+package prometheus
+
+import (
+	"compress/gzip"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/segmentio/stats"
+)
+
+// Handler is a type that bridges the stats API to a prometheus-compatible HTTP
+// endpoint.
+//
+// Typically, a program creates one Handler, registers it to the stats package,
+// and adds it to the muxer used by the application under the /metrics path.
+type Handler struct {
+	// A map of metric name to histograms buckets used by the handler to record
+	// observed values.
+	//
+	// The buckets are of list of upper limits used to group the observed
+	// values.
+	//
+	// If a bucket is missing for a metric name the observed value is ignored.
+	HistgramBuckets map[string][]float64
```

Comment:
(also I just noticed the typo _Histgram_ => _Histogram_, I'll fix later otherwise github will drop the comment)